lib/SPIRV/libSPIRV/SPIRVBasicBlock.cpp

```diff
@@ -87,6 +87,17 @@ void SPIRVBasicBlock::encodeChildren(spv_ostream &O) const {
 
 _SPIRV_IMP_ENCDEC1(SPIRVBasicBlock, Id)
 
+SPIRVInstruction *SPIRVBasicBlock::getVariableInsertionPoint() const {
+  auto IP =
+      std::find_if(InstVec.begin(), InstVec.end(), [](SPIRVInstruction *Inst) {
```

Comment:
@victor-eds there is an issue with the patch, in situations when we have phi instruction and alloca within the same basic block - OpVariable must not be placed before OpPhi. May I ask you to fix that?

Reply:
That would be illegal SPIR-V anyway, but I can issue a fix.

Reply:
Why? The SPIR-V spec says:
```
Within a block, this instruction must appear before all non-OpPhi instructions
(except for OpLine and OpNoLine, which can be mixed with OpPhi).
```

Reply:
I wonder if this is an overlooked edge case and needs a spec fix. Normally with structured spirv you'll never run into a situation where you'll an `OpVariable` and an `OpPhi` instruction in the same block, but with OpenCL SPIR-V this can happen in the very first one.